### ISSUE TITLE
Feature request: Jupyterlab services provide a way to retrieve stdout from a connection

### ISSUE THREAD

**Issue by rchiodo** [jupyterlab/jupyterlab#7904]
[We](https://github.com/Microsoft/vscode-python) use the jupyterlab services to connect to either jupyter notebook servers on the local machine or a remote machine. 

In the local case we surface the stdout/stderr from the local process to VS code for the user to see.

We'd love to be able to do the same for the remote case. Sorry if this is already possible.

**Comment by vidartf** [comment]
I'm unsure what you mean with "stdout/stderr from the local process". Do you mean the streams from the server process? If so, that is not directly available via the Jupyter protocols, and might be considered a security issue.

**Comment by rchiodo** [comment]
Yes the stdout that the remote server process is outputting. I'm not sure how that would be a security problem as the user has a kernel running on the remote machine in this situation anyway which lets them execute code there anyway. This is certainly less of a problem than having code execute.

The problem we're trying to solve is that the stdout from the jupyter server (and stderr) include useful information that might relate to what the user is doing. Like say if a kernel is unable to find a dll in order to run. This information isn't surfaced in the response, just that the kernel died.

**Comment by jasongrout** [comment]
The short answer to your question is that there is no mechanism on the server right now for communicating the server log to the frontend. This would need to change on the server side first to make those logs available to a frontend, so a PR should be opened in the [notebook server repo](https://github.com/jupyter/notebook/) (or the upcoming [jupyter server repo](https://github.com/jupyter/jupyter_server)?)

**Comment by jasongrout** [comment]
And perhaps only kernel-related log messages would be available to the frontend?

**Comment by rchiodo** [comment]
> And perhaps only kernel-related log messages would be available to the frontend?

Yeah that seems sufficient to me. We don't actually need all of the stdout put out by the server. Really just the secondary information once it's already started.
